src/libraries/System.Text.Json/gen/Reflection/TypeWrapper.cs

```diff
@@ -281,6 +281,11 @@ public override Type GetElementType()
             return _elementType;
         }
 
+        public override Type MakeArrayType()
+        {
+            return _metadataLoadContext.Compilation.CreateArrayTypeSymbol(_typeSymbol).AsType(_metadataLoadContext);
```

Comment:
I see. Originally I thought (without looking at the code) that we didn't add the converter for `typeof(byte[])`, but the issue is actually related to Roslyn special treatment of arrays.